root/static/scripts/edit/URLCleanup.js

```diff
@@ -635,6 +658,23 @@ const CLEANUPS = {
       return url;
     },
     validate: function (url, id) {
+      if (/apple\.co\//i.test(url)) {
+        return {
+          error: exp.l(
+            `This is a redirect link. Please follow {redirect_url|your link}
+             and add the link it redirects to instead.`,
+            {
+              redirect_url: {
+                href: url,
+                rel: 'noopener noreferrer',
+                target: '_blank',
+              },
+            },
+          ),
+          result: false,
+        };
+      }
+
```

Comment:
Frustratingly, given we don't autoselect a type but just clean up the URL at the moment, there's no way (AFAICT) to keep this validation in the current code (validation is only ran `if (cleanup && cleanup.type && cleanup.type[entityType])`. Should we add an id-agnostic validation option for cases like this? Or should we just special-case this one?

Reply:
The real issue is MBS-9902. I would just leave it as-is in the meantime.

Reply:
As is meaning "don't block Apple redirect links for now"? (AFAICT I can't block them normally without it going through validation)

Reply:
`apple.co` redirect links can be blocked by creating a separate `'appleco'` rule ahead of `'applemusic'`.

Reply:
I went back to use `applemusic` now since we can have validation there.